runtests.c

```diff
@@ -0,0 +1,236 @@
+#include "volcano.h"
+
+int testcurl() {
+  printf("%s\n",__FUNCTION__);
+
+  char* buf;
+  long len = download("https://dl.ash2txt.org/full-scrolls/Scroll1/PHercParis4.volpkg/paths/20230503225234/author.txt", &buf);
+  if(len!= 6) {
+    return 1;
+  }
+  if (strncmp(buf,"noemi",5) != 0) {return 1;}
+  free(buf);
+  return 0;
+}
+
+int testhistogram() {
+  printf("%s\n",__FUNCTION__);
+
+  chunk* mychunk = tiff_to_chunk("./example_data/example_3d.tif");
+  slice* myslice = tiff_to_slice("./example_data/example_3d.tif", 0);
+  histogram* slice_hist = slice_histogram(myslice->data, myslice->dims[0],myslice->dims[1], 256);
+  histogram* chunk_hist = chunk_histogram(mychunk->data, mychunk->dims[0],mychunk->dims[1],mychunk->dims[2], 256);
+
+  hist_stats stats = calculate_histogram_stats(slice_hist);
+  printf("Mean: %.2f\n", stats.mean);
+  printf("Median: %.2f\n", stats.median);
+  printf("Mode: %.2f (count: %u)\n", stats.mode, stats.mode_count);
+  printf("Standard Deviation: %.2f\n", stats.std_dev);
+
+  write_histogram_to_csv(slice_hist, "slice_histogram.csv");
+  write_histogram_to_csv(chunk_hist, "chunk_histogram.csv");
+
+  histogram_free(slice_hist);
+  histogram_free(chunk_hist);
+  chunk_free(mychunk);
+  slice_free(myslice);
+  return 0;
+}
+
+int testzarr() {
+  printf("zarr support not yet implemented\n");
+  return 0;
+  printf("%s\n",__FUNCTION__);
```

Comment:
I would favour commenting this block out, rather than having the return 0; do all the work.